unittests/basic_tests.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+#** *****************************************************************************
+# *
+# * If not stated otherwise in this file or this component's LICENSE file the
+# * following copyright and licenses apply:
+# *
+# * Copyright 2024 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import contextlib
+import io
+import os
+from os import path
+import shutil
+import sys
+import unittest
+
+import yaml
+try:
+	from yaml import CSafeLoader as SafeLoader
+except ImportError:
+	from yaml import SafeLoader
+
+MY_PATH = path.abspath(__file__)
+MY_DIR = path.dirname(MY_PATH)
+
+sys.path.append(path.join(MY_DIR,'../src'))
+from yaml_runner import YamlRunner
+
+YAMLRUNNER_PATH = path.abspath(path.join(MY_DIR,'../source/yaml_runner.py'))
+
+class BasicTest(unittest.TestCase):
+    workspace = path.join(MY_DIR,'workspace')
+    test_config_path = path.join(MY_DIR,'../examples/test_config.yml')
+    test_config_file = open(test_config_path, 'r', encoding='utf-8')
+    test_dict = yaml.load(test_config_file, SafeLoader)
+    test_configs = {'path': test_config_path,
+                    'dict' : test_dict,
+                    'file handle' : test_config_file}
+    stderr_log = None
+    stdout_log = None
+    runner = YamlRunner('BasicTest')
+
+    @classmethod
+    def setUpClass(cls):
+        if path.exists(cls.workspace):
+            shutil.rmtree(cls.workspace)
+        os.mkdir(cls.workspace)
+
+    @classmethod
+    def setUp(cls):
+        cls.stderr_log = io.StringIO()
+        cls.stdout_log = io.StringIO()
+
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        shutil.rmtree(cls.workspace)
+        cls.test_config_file.close()
```

Comment:
if cls.test_config_file and not cls.test_config_file.closed:
        cls.test_config_file.close()